libs/api/domains/occupational-licenses/src/lib/occupationalLicenses.service.ts

```diff
@@ -1,286 +1,293 @@
+import { LicenseType } from './models/licenseType.model'
+import { LicenseResponse as MMSLicenseResponse } from '@island.is/clients/mms'
+import { Status } from './models/licenseStatus.model'
+import { LicenseError } from './models/licenseError.model'
+import { FetchError } from '@island.is/clients/middlewares'
+import { m } from './messages'
 
-type OccupationalLicenseResult<T> =
-  | {
-      items: T[]
-      type: 'data'
-      error?: never
-    }
-  | {
-      items?: never
-      error: OccupationalLicensesError
-      type: 'error'
-    }
+const NAMESPACE_ID = 'api.occupational-licenses'
 
-@Injectable()
 export class OccupationalLicensesService {
   constructor(
-    private healthDirectorateApi: HealthDirectorateClientService,
-    private mmsApi: MMSApi,
+    private readonly dcService: DistrictCommissionersLicensesService,
+    private readonly healthService: HealthDirectorateClientService,
+    private readonly mmsApi: MMSApi,
+    private readonly intlService: IntlService,
     @Inject(DownloadServiceConfig.KEY)
     private readonly downloadService: ConfigType<typeof DownloadServiceConfig>,
-
-    private readonly featureFlagService: FeatureFlagService,
-    @Inject(LOGGER_PROVIDER)
-    private logger: Logger,
   ) {}
+  async getDistrictCommissionerLicenses(
+    user: User,
+  ): Promise<Array<License> | LicenseError | null> {
+    const licenses = await this.dcService
+      .getLicenses(user)
+      .catch((e: Error | FetchError) => {
+        return {
+          type: LicenseType.DISTRICT_COMMISSIONERS,
+          error: JSON.stringify(e),
+        }
+      })
 
-  private checkHealthDirectorateValidity = (
-    status: HealthDirectorateLicenseStatus,
-  ): OccupationalLicenseStatus => {
-    switch (status) {
-      case 'VALID':
-        return OccupationalLicenseStatus.valid
-      case 'LIMITED':
-        return OccupationalLicenseStatus.limited
-      case 'INVALID':
-        return OccupationalLicenseStatus.error
-      case 'REVOKED':
-        return OccupationalLicenseStatus.revoked
-      case 'WAIVED':
-        return OccupationalLicenseStatus.waived
-      default:
-        this.logger.warn('Unknown health directorate status', {
-          category: LOG_CATEGORY,
-          status: status,
-        })
-        return OccupationalLicenseStatus.error
+    if (!licenses) {
+      return []
     }
-  }
 
-  private checkEducationalValidity = (
-    validFrom: string,
-  ): OccupationalLicenseStatus => {
-    return new Date(validFrom) < new Date()
-      ? OccupationalLicenseStatus.valid
-      : OccupationalLicenseStatus.error
-  }
+    if (Array.isArray(licenses)) {
+      const data: Array<License> =
+        licenses
+          ?.map((l) => {
+            if (!l || !l.status || !l.title || !l.validFrom) {
+              return
+            }
+            return {
+              licenseId: addLicenseTypePrefix(l.id, 'DistrictCommissioners'),
+              type: LicenseType.DISTRICT_COMMISSIONERS,
```

Comment:
Or is it possible to use the type from LicenseType?

Reply:
oh yeah that's more efficient